sdk/core/Azure.Core/tests/RequestActivityPolicyTests.cs

```diff
@@ -61,9 +61,9 @@ public async Task ActivityIsCreatedForRequest()
             CollectionAssert.Contains(activity.Tags, new KeyValuePair<string, string>("http.user_agent", "agent"));
             CollectionAssert.Contains(activity.Tags, new KeyValuePair<string, string>("requestId", clientRequestId));
             CollectionAssert.Contains(activity.Tags, new KeyValuePair<string, string>("serviceRequestId", "server request id"));
+            CollectionAssert.Contains(activity.Tags, new KeyValuePair<string, string>("otel.status_code", "UNSET"));
```

Comment:
Do we have a test that validates it being set to "ERROR"?

Reply:
We do have one https://github.com/pakrym/azure-sdk-for-net/blob/pakrym/Always-set-otel-status_code/sdk/core/Azure.Core/tests/RequestActivityPolicyTests.cs#L87